src/main/java/com/limechain/trie/structure/decoded/node/DecodedNode.java

```diff
@@ -228,4 +240,221 @@ public byte[] calculateMerkleValue(Function<byte[], byte[]> hashFunction, boolea
 
         return nodeValue;
     }
+
+    /***
+     * Decodes a node value found in a proof into its components.
+     * This can decode nodes no matter their version or hash algorithm.
+     *
+     * @param encoded The byte array representing the encoded node.
+     * @return DecodedNode<Nibbles, List < Byte>> The decoded node with its partial key, children, and storage value.
+     * @throws NodeDecodingException If the encoded array is null, empty, or invalid.
+     * @throws IllegalArgumentException If 'encoded' is null.
+     */
+    public static DecodedNode decode(byte[] encoded) {
+        ScaleCodecReader reader = new ScaleCodecReader(encoded);
+        if (encoded == null || encoded.length == 0) {
+            throw new NodeDecodingException("Invalid node value: it's empty");
+        }
+
+        int firstByte = reader.readByte() & 0xFF;
+        boolean hasChildren;
+        Boolean storageValueHashed;
+        int pkLenFirstByteBits;
+
+        //https://spec.polkadot.network/#defn-node-header
```

Comment:
Let's keep code style consistent and extract parts of the decode logic into separate functions just like in `encode` above.
<img width="719" alt="image" src="https://github.com/LimeChain/Fruzhin/assets/29047760/fca7872c-7e19-4a0e-a0fe-98bcb795defd">

Reply:
We reached the conclusion it's okay as is, as far as possible what's extractable has been extracted.